src/protocols/http/hpr_http_roaming_worker.erl

```diff
@@ -0,0 +1,258 @@
+%%%-------------------------------------------------------------------
+%%% @author jonathanruttenberg
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 21. Sep 2022 1:01 PM
+%%%-------------------------------------------------------------------
+-module(hpr_http_roaming_worker).
+-author("jonathanruttenberg").
+
+-behavior(gen_server).
+
+-include("hpr_http_roaming.hrl").
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+-export([
+    start_link/1,
+    handle_packet/4
+]).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Exports
+%% ------------------------------------------------------------------
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2
+]).
+
+-define(SERVER, ?MODULE).
+-define(SEND_DATA, send_data).
+-define(SHUTDOWN, shutdown).
+-define(SHUTDOWN_TIMEOUT, timer:seconds(5)).
+
+-type address() :: binary().
+
+-record(state, {
+    net_id :: hpr_http_roaming:netid_num(),
+    address :: address(),
+    transaction_id :: integer(),
+    packets = [] :: list(hpr_http_roaming:packet()),
+
+    send_data_timer = 200 :: non_neg_integer(),
+    send_data_timer_ref :: undefined | reference(),
+    flow_type :: async | sync,
+    auth_header :: null | binary(),
+    protocol_version :: pv_1_0 | pv_1_1,
+
+    should_shutdown = false :: boolean(),
+    shutdown_timer_ref :: undefined | reference()
+}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Args) ->
+    gen_server:start_link(?SERVER, Args, []).
+
+-spec handle_packet(
+    WorkerPid :: pid(),
+    PacketUp :: hpr_packet_up:packet(),
+    GatewayTime :: hpr_http_roaming:gateway_time(),
+    GatewayStream :: hpr_router_stream_manager:gateway_stream()
+) -> ok | {error, any()}.
+handle_packet(Pid, PacketUp, GatewayTime, GatewayStream) ->
+    lager:debug(
```

Comment:
I don't know if we need this log. 
Most of this information has been logged a few times by now. 
And logging full packets gets very noisy, and the log aggregator doesn't like it.

Reply:
Since it is _debug_, it is not routinely logged.
However, I can remove it if you still think it is a problem.

Reply:
I have removed it.